src/main/java/net/snowflake/ingest/connection/ServiceResponseHandler.java

```diff
@@ -167,10 +164,31 @@ private static void handleExceptionalStatus(StatusLine statusLine, HttpResponse
         // We don't know how to respond now...
       default:
         LOGGER.error("Status code {} found in response from service", statusLine.getStatusCode());
-        String blob = EntityUtils.toString(response.getEntity());
+        String blob = consumeAndReturnResponseEntityAsString(response.getEntity());
         throw new IngestResponseException(
             statusLine.getStatusCode(),
             IngestResponseException.IngestExceptionBody.parseBody(blob));
     }
   }
+
+  /**
+   * Consumes the HttpEntity as mentioned in <a
+   * href="https://hc.apache.org/httpcomponents-client-4.5.x/quickstart.html">HttpClient Docs</a>
+   *
+   * <p>Also returns the string version of this entity.
+   *
+   * @param httpResponseEntity the response entity obtained after successfully calling associated
+   *     Rest APIs
+   * @return String version of this http response which will be later used to deserialize into
+   *     respective Response Object
+   * @throws IOException if parsing error
+   */
+  private static String consumeAndReturnResponseEntityAsString(HttpEntity httpResponseEntity)
+      throws IOException {
+    // grab the string version of the response entity
+    String responseEntityAsString = EntityUtils.toString(httpResponseEntity);
+
+    EntityUtils.consumeQuietly(httpResponseEntity);
```

Comment:
Looks like we will swallow the exception if using `consumeQuietly`?

Reply:
Yess, we will.. But this was released in 0.10.6